--- a/cunit/timeofday.c
+++ b/cunit/timeofday.c
@@ -186,32 +186,18 @@
  * Platform-specific libc interception code
  */

-#if defined(__GLIBC__)
-
 /* call the real libc function */
 static int real_gettimeofday(struct timeval *tv, ...)
 {
-    extern int __gettimeofday(struct timeval *, ...);
-    return __gettimeofday(tv, NULL);
+    return gettimeofday(tv, NULL);
 }

-/* provide a function to hide the libc weak alias */
-int gettimeofday(struct timeval *tv, ...)
-{
-    to_timeval(transform(now()), tv);
-    return 0;
-}
-
 time_t time(time_t *tp)
 {
     time_t tt = to_time_t(transform(now()));
     if (tp) *tp = tt;
     return tt;
 }
-
-#else
-#error "Don't know how to intercept gettimeofday for this libc"
-#endif


 /*
